src/content/en/updates/2018/08/wasm-av1.md

```diff
@@ -0,0 +1,311 @@
+
+{# wf_published_on: 2018-08-21 #}
+{# wf_updated_on: 2018-08-21 #}
+{# wf_tags: webassembly,media #}
+{# wf_featured_image: /web/updates/images/generic/play-outline.png #}
+{# wf_featured_snippet: WebAssembly lets us extend the browser with new features. This article shows how to port the AV1 video decoder and play AV1 video in any modern browser. #}
+{# wf_blink_components: Blink #}
+
+# Extending the browser with WebAssembly {: .page-title }
+
+{% include "web/_shared/contributors/alexdanilo.html" %}
+
+One of the best things about [WebAssembly](https://webassembly.org/) is the ability experiment with new capabilities and implement new ideas before the browser ships those features natively (if at all).. You can think of using WebAssembly this way as a high-performance polyfill mechanism, where you write your feature in C/C++ or Rust rather than JavaScript.
+
+With a plethora of existing code available for porting, it's possible to do things in the browser that weren't viable until WebAssembly came along.
+
+This article will walk through an example of how to take the existing AV1 video codec source code, build a wrapper for it, and try it out inside your browser and tips to help with building a test harness to debug the wrapper. Full source code for the example here is available at https://github.com/GoogleChromeLabs/wasm-av1 for reference.
+
+
+## Choosing an interesting code-base
+
+For a number of years now, we've seen that a large percentage of traffic on the web consists of video data, Cisco [estimates it](https://www.cisco.com/c/en/us/solutions/collateral/service-provider/visual-networking-index-vni/complete-white-paper-c11-481360.html#_Toc484813971) as much as 80% in fact! Of course, browser vendors and video sites are hugely aware of the desire to reduce the data consumed by all this video content. The key to that, of course, is better compression, and as you'd expect there is a lot of research into next-generation video compression aimed at reducing the data burden of shipping video across the internet.
+
+As it happens, the [Alliance for Open Media](https://aomedia.org/) has been working on a next generation video compression scheme called [AV1](https://aomedia.org/av1-features/get-started/) that promises to shrink video data size considerably. In the future, we'd expect browsers to ship native support for AV1, but luckily the source code for the compressor and decompressor are [open source](https://aomedia.googlesource.com/aom/), which makes that an ideal candidate for trying to compile it into WebAssembly so we can experiment with it in the browser.
+
+
+## Adapting for use in the browser
+
+One of the first things we need to do to get this code into the browser is to get to know the existing code to understand what the API is like. When first looking at this code, two things stand out:
+
+
+
+1.  The source tree is built using a tool called **<code>cmake</code>**; and
```

Comment:
Minor style nit: can you just use backticks instead of `<code>`? Not a big deal.